### PR TITLE
Add Bns executable with control systems

### PR DESCRIPTION
**Pull request by knelli2**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Executables/GrMhd/GhValenciaDivClean/CMakeLists.txt [hunk 1]
```diff
@@ -52,43 +52,57 @@ set(LIBS_TO_LINK
...
+  target_compile_definitions(
+    EvolveGhValenciaDivCleanBns PRIVATE HAS_SPEC_EXPORTER)
 endif()
 
 add_ghmhd_with_horizon_executable(
   BondiHoyleAccretion
-  gh::Solutions::WrappedGr<grmhd::AnalyticData::BondiHoyleAccretion>
-  "${LIBS_TO_LINK}"
+  "gh::Solutions::WrappedGr<grmhd::AnalyticData::BondiHoyleAccretion>,false"
+  "${LIBS_TO_LINK};ApparentHorizons;ParallelInterpolation"
```
Comment (ffoucart) [inline src/Evolution/Executables/GrMhd/GhValenciaDivClean/CMakeLists.txt:83]: Is the addition of ApparentHorizons supposed to be part of this commit?

File: tests/InputFiles/GrMhd/GhValenciaDivClean/BinaryNeutronStar.yaml [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
   AvoidGlobalProc0: false
+  Singletons: Auto
 
 Evolution:
   InitialTime: 0.0
-  # The initial time step gets overriden by `ChangeSlabSize` below
+  # The initial time step gets overridden by `ChangeSlabSize` below
   InitialTimeStep: 0.075
+  # Control systems only work with AdamsBashforth or AdamsMoultonPc
   TimeStepper: Rk3HesthavenSsp
```
Comment (ffoucart) [inline tests/InputFiles/GrMhd/GhValenciaDivClean/BinaryNeutronStar.yaml]: You probably want AB3 here.

**Review by ffoucart — COMMENTED** [review]
I need to look at the main executable as well... is this something that now should be able to run our BNS setup (i.e. this branch + develop should have everything we need?)

**Comment by nilsdeppe** [comment]
I'll need to update this with the starting on FD code to get BNS working, but I think we can merge either first and just combine after :)

**Comment by ffoucart** [comment]
That's ok though; I should be able to start a test run with the ID interpolated on the Dg grid first. It won't be as good as our latest runs, but it should at least work...

**Comment by nilsdeppe** [comment]
Yea, and I'm hoping to have a PR for most of my changes submitted in a couple hours. It's just finishing CI on my branch. There'll be one more PR after that to get starting on FD working, but it's getting close.

File: src/Evolution/Executables/GrMhd/GhValenciaDivClean/CMakeLists.txt (hunk 1 above)
Comment (knelli2) [reply src/Evolution/Executables/GrMhd/GhValenciaDivClean/CMakeLists.txt:83]: Yeah for some reason I got linking errors with these with my changes so I had to add them in.

File: tests/InputFiles/GrMhd/GhValenciaDivClean/BinaryNeutronStar.yaml (hunk 2 above)
Comment (knelli2) [reply tests/InputFiles/GrMhd/GhValenciaDivClean/BinaryNeutronStar.yaml]: Yeah I can change the default

File: src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp [hunk 3]
```diff
@@ -516,21 +558,31 @@ struct GhValenciaDivCleanTemplateBase<
...
+                       domain_creators<volume_dim>>>,
         tmpl::pair<
             Event,
             tmpl::flatten<tmpl::list<
                 Events::Completion,
-                dg::Events::field_observations<volume_dim, Tags::Time,
-                                               observe_fields,
-                                               non_tensor_compute_tags>,
+                dg::Events::field_observations<
+                    volume_dim, Tags::Time, observe_fields,
```
Comment (nilsdeppe) [inline src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp]: How does this compile? `Tags::Time` should be the list of fields when I look at the definition. What am I missing?

Edit: I think this branch isn't rebased and tested on the latest develop and these tags were removed

**Comment by knelli2** [comment]
Ok I have added two more commits before the one that changes the metavars. These commits have the control system event/measurements get tags from the ObservationBox rather than the DataBox. This simplified changes in the metavars quite a bit. @wthrowe do you mind taking a look at the first two commits?

File: tests/InputFiles/GrMhd/GhValenciaDivClean/BinaryNeutronStar.yaml [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
   AvoidGlobalProc0: false
+  Singletons: Auto
 
 Evolution:
   InitialTime: 0.0
-  # The initial time step gets overriden by `ChangeSlabSize` below
+  # The initial time step gets overridden by `ChangeSlabSize` below
   InitialTimeStep: 0.075
-  TimeStepper: Rk3HesthavenSsp
+  # Control systems only work with AdamsBashforth or AdamsMoultonPc
```
Comment (wthrowe) [inline tests/InputFiles/GrMhd/GhValenciaDivClean/BinaryNeutronStar.yaml]: No AdamsMoultonPc.

**Review by wthrowe — CHANGES_REQUESTED** [review]
First two commits look fine.

**Review by wthrowe — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**